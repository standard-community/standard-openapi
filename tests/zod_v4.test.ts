import { describe, expect, it } from "vitest";
import z from "zod/v4";

import { toOpenAPISchema } from "~/index.js";

describe("zod v4", () => {
  it("basic", async () => {
    const schema = z
      .object({
        myString: z.string(),
        myUnion: z.union([z.number(), z.boolean()]),
      })
      .describe("My neat object schema");

    const specs = await toOpenAPISchema(schema);
    expect(specs).toMatchSnapshot();
  });

  it("with metadata", async () => {
    const schema = z
      .object({
        myString: z.string(),
        myUnion: z.union([z.number(), z.boolean()]),
      })
      .describe("My neat object schema").meta({
        ref: "MyNeatObjectSchema",
      });

    const specs = await toOpenAPISchema(schema);
    expect(specs).toMatchSnapshot();
  });
});
