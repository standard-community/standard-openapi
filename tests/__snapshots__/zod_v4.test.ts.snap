// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`zod v4 > basic 1`] = `
{
  "components": undefined,
  "schema": {
    "additionalProperties": false,
    "description": "My neat object schema",
    "properties": {
      "myString": {
        "type": "string",
      },
      "myUnion": {
        "anyOf": [
          {
            "type": "number",
          },
          {
            "type": "boolean",
          },
        ],
      },
    },
    "required": [
      "myString",
      "myUnion",
    ],
    "type": "object",
  },
}
`;

exports[`zod v4 > with metadata 1`] = `
{
  "components": {
    "schemas": {
      "MyNeatObjectSchema": {
        "additionalProperties": false,
        "description": "My neat object schema",
        "properties": {
          "myString": {
            "type": "string",
          },
          "myUnion": {
            "anyOf": [
              {
                "type": "number",
              },
              {
                "type": "boolean",
              },
            ],
          },
        },
        "required": [
          "myString",
          "myUnion",
        ],
        "type": "object",
      },
    },
  },
  "schema": {
    "$ref": "#/components/schemas/MyNeatObjectSchema",
  },
}
`;
